File: Python/two-sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Time:  O(n)
        # Space: O(n)
        m = {}
        for i, val in enumerate(nums):
            diff = target - val
            if diff in m:
                return i, m[diff]
            m[val] = i

File: C++/two-sum.cpp
#include <vector>
#include <map>


class Solution {
public:
    std::vector<int> twoSum(std::vector<int> nums, int target) {
        // Time:  O(n)
        // Space: O(n)
        std::vector<int> result;
        std::map<int, int> m;
        for (int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];
            auto it = m.find(diff);
            if (it != m.end()) {
                result = {it->second, i};
                break;
            } else {
                m[nums[i]] = i;
            }
        }
        return result;
    }
};